### PR TITLE
feat: custom domain front end 

### PR DESCRIPTION
**Pull request by Kneesal**
# Description

copilot:summary

- Link to Basecamp Todo

# How should this PR be QA Tested?

Please describe the QA tests that you ran to verify your changes. Please also note any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

# Walkthrough

copilot:walkthrough

### CODE REVIEW

**Review by mikeallisonJS — DISMISSED** [review]

**Comment by tataihono** [comment]
atlantis unlock

**Comment by tataihono** [comment]
atlantis unlock

File: apps/journeys-admin/pages/api/preview.ts [hunk 1]
```diff
@@ -53,5 +55,10 @@ export default async function handler(
   // 300ms required to invalidate edge caches
   await sleep(300)
 
-  res.redirect(307, `${process.env.JOURNEYS_URL}/${slug}`)
+  const proto = process.env.NODE_ENV === 'development' ? 'http://' : 'https://'
+
+  res.redirect(
+    307,
+    `${hostname != null ? proto + hostname : process.env.JOURNEYS_URL}/${slug}`
```
Comment (tataihono) [inline apps/journeys-admin/pages/api/preview.ts:62]:
```suggestion
    `${hostname != null ? `${proto}${hostname}` : process.env.JOURNEYS_URL}/${slug}`
```

File: apps/journeys-admin/src/components/Editor/EditToolbar/EditToolbar.tsx [hunk 2]
```diff
@@ -20,6 +22,12 @@ export function EditToolbar(): ReactElement {
   const { t } = useTranslation('apps-journeys-admin')
   const { journey } = useJourney()
   const { state } = useEditor()
+  const { data: customDomains } = useCustomDomainsQuery({
+    variables: { teamId: journey?.team?.id as string }
+  })
+
+  const hostname = new URL(`https://${customDomains?.customDomains[0].name}`)
```
Comment (tataihono) [inline apps/journeys-admin/src/components/Editor/EditToolbar/EditToolbar.tsx:29]:
```suggestion
  const hostname = customDomains?.customDomains[0].name
```

File: apps/journeys-admin/src/components/Editor/EditToolbar/EditToolbar.tsx [hunk 3]
```diff
@@ -35,7 +43,7 @@ export function EditToolbar(): ReactElement {
             icon={<EyeOpenIcon />}
             label={t('Preview')}
             component="a"
-            href={`/api/preview?slug=${journey.slug}`}
+            href={`/api/preview?slug=${journey.slug}&hostname=${hostname}`}
```
Comment (tataihono) [inline apps/journeys-admin/src/components/Editor/EditToolbar/EditToolbar.tsx:46]:
```suggestion
            href={`/api/preview?slug=${journey.slug}${hostname ? `&hostname=${hostname}`: ''}`}
```

File: apps/journeys-admin/src/components/Editor/EditToolbar/EditToolbar.tsx [hunk 4]
```diff
@@ -48,7 +56,7 @@ export function EditToolbar(): ReactElement {
           />
           <IconButton
             aria-label="Preview"
-            href={`/api/preview?slug=${journey.slug}`}
+            href={`/api/preview?slug=${journey.slug}&hostname=${hostname}`}
```
Comment (tataihono) [inline apps/journeys-admin/src/components/Editor/EditToolbar/EditToolbar.tsx:59]:
```suggestion
            href={`/api/preview?slug=${journey.slug}${hostname ? `&hostname=${hostname}`: ''}`}
```

File: apps/journeys-admin/src/components/Editor/EditToolbar/Menu/Menu.tsx [hunk 5]
```diff
@@ -81,6 +86,9 @@ export function Menu(): ReactElement {
   >()
   const [anchorEl, setAnchorEl] = useState<HTMLElement | null>(null)
 
+  const hostname = new URL(`https://${customDomains?.customDomains[0].name}`)
+    .hostname
```
Comment (tataihono) [inline apps/journeys-admin/src/components/Editor/EditToolbar/Menu/Menu.tsx:90]:
```suggestion
  const hostname = customDomains?.customDomains[0].name
```

File: apps/journeys-admin/src/components/Editor/EditToolbar/Menu/Menu.tsx [hunk 6]
```diff
@@ -146,7 +154,7 @@ export function Menu(): ReactElement {
         }}
       >
         <NextLink
-          href={`/api/preview?slug=${journey?.slug ?? ''}`}
+          href={`/api/preview?slug=${journey?.slug ?? ''}&hostname=${hostname}`}
```
Comment (tataihono) [inline apps/journeys-admin/src/components/Editor/EditToolbar/Menu/Menu.tsx:157]:
```suggestion
          href={`/api/preview?slug=${journey?.slug ?? ''}${hostname ? `&hostname=${hostname}`: ''}`}
```

File: apps/journeys-admin/src/components/Editor/Properties/JourneyLink/EmbedJourneyDialog/EmbedJourneyDialog.spec.tsx
Comment (tataihono) [inline apps/journeys-admin/src/components/Editor/Properties/JourneyLink/EmbedJourneyDialog/EmbedJourneyDialog.spec.tsx:1]: revert

File: apps/journeys-admin/src/components/Editor/Properties/JourneyLink/EmbedJourneyDialog/EmbedJourneyDialog.stories.tsx
Comment (tataihono) [inline apps/journeys-admin/src/components/Editor/Properties/JourneyLink/EmbedJourneyDialog/EmbedJourneyDialog.stories.tsx:1]: revert

File: apps/journeys-admin/src/components/Editor/Properties/JourneyLink/EmbedJourneyDialog/EmbedJourneyDialog.tsx
Comment (tataihono) [inline apps/journeys-admin/src/components/Editor/Properties/JourneyLink/EmbedJourneyDialog/EmbedJourneyDialog.tsx:1]: revert

File: apps/journeys-admin/src/components/Editor/Properties/JourneyLink/JourneyLink.tsx [hunk 7]
```diff
@@ -33,6 +38,9 @@ export function JourneyLink(): ReactElement {
     })
   }
 
+  const customDomainVerified: boolean =
+    customDomains != null && customDomains.customDomains.length > 0
```
Comment (tataihono) [inline apps/journeys-admin/src/components/Editor/Properties/JourneyLink/JourneyLink.tsx:42]:
```suggestion
  const hostname = customDomains?.customDomains[0].name
```

File: apps/journeys-admin/src/components/Editor/Properties/JourneyLink/JourneyLink.tsx [hunk 8]
```diff
@@ -44,8 +52,10 @@ export function JourneyLink(): ReactElement {
         value={
           journey?.slug != null
             ? `${
-                process.env.NEXT_PUBLIC_JOURNEYS_URL ??
-                'https://your.nextstep.is'
+                customDomainVerified
+                  ? 'https://' + customDomains?.customDomains[0].name
+                  : process.env.NEXT_PUBLIC_JOURNEYS_URL ??
+                    'https://your.nextstep.is'
```
Comment (tataihono) [inline apps/journeys-admin/src/components/Editor/Properties/JourneyLink/JourneyLink.tsx:58]: use hostname

File: apps/journeys-admin/src/components/Editor/Properties/JourneyLink/JourneyLink.tsx [hunk 9]
```diff
@@ -87,10 +97,12 @@ export function JourneyLink(): ReactElement {
       <SlugDialog
         open={showSlugDialog}
         onClose={() => setShowSlugDialog(false)}
+        customDomainName={customDomains?.customDomains[0]?.name}
```
Comment (tataihono) [inline apps/journeys-admin/src/components/Editor/Properties/JourneyLink/JourneyLink.tsx:100]:
```suggestion
        hostname={hostname}
```

File: apps/journeys-admin/src/components/Editor/Properties/JourneyLink/JourneyLink.tsx [hunk 10]
```diff
@@ -87,10 +97,12 @@ export function JourneyLink(): ReactElement {
...
       />
       <EmbedJourneyDialog
         open={showEmbedDialog}
         onClose={() => setShowEmbedDialog(false)}
+        customDomainName={customDomains?.customDomains[0]?.name}
```
Comment (tataihono) [inline apps/journeys-admin/src/components/Editor/Properties/JourneyLink/JourneyLink.tsx:105]:
```suggestion
        hostname={hostname}
```

File: apps/journeys-admin/src/components/Editor/Properties/JourneyLink/SlugDialog/SlugDialog.tsx [hunk 11]
```diff
@@ -23,9 +23,14 @@ export const JOURNEY_SLUG_UPDATE = gql`
 interface SlugDialogProps {
   open: boolean
   onClose: () => void
+  customDomainName: string | undefined
```
Comment (tataihono) [inline apps/journeys-admin/src/components/Editor/Properties/JourneyLink/SlugDialog/SlugDialog.tsx:26]:
```suggestion
  hostname: string
```

File: apps/journeys-admin/src/components/Editor/Properties/JourneyLink/SlugDialog/SlugDialog.tsx [hunk 12]
```diff
@@ -23,9 +23,14 @@ export const JOURNEY_SLUG_UPDATE = gql`
...
 }
 
-export function SlugDialog({ open, onClose }: SlugDialogProps): ReactElement {
+export function SlugDialog({
+  open,
+  onClose,
+  customDomainName
```
Comment (tataihono) [inline apps/journeys-admin/src/components/Editor/Properties/JourneyLink/SlugDialog/SlugDialog.tsx:32]:
```suggestion
  hostname
```

File: apps/journeys-admin/src/components/Editor/Properties/JourneyLink/SlugDialog/SlugDialog.tsx [hunk 13]
```diff
@@ -74,6 +79,8 @@ export function SlugDialog({ open, onClose }: SlugDialogProps): ReactElement {
     }
   }
 
+  const hasCustomDomain: boolean = customDomainName != null
```
Comment (tataihono) [inline apps/journeys-admin/src/components/Editor/Properties/JourneyLink/SlugDialog/SlugDialog.tsx:82]: drop

File: apps/journeys-admin/src/components/Editor/Properties/JourneyLink/SlugDialog/SlugDialog.tsx [hunk 14]
```diff
@@ -106,8 +113,10 @@ export function SlugDialog({ open, onClose }: SlugDialogProps): ReactElement {
                   helperText={
                     values.slug !== '' ? (
                       <>
-                        {process.env.NEXT_PUBLIC_JOURNEYS_URL ??
-                          'https://your.nextstep.is'}
+                        {hasCustomDomain
```
Comment (tataihono) [inline apps/journeys-admin/src/components/Editor/Properties/JourneyLink/SlugDialog/SlugDialog.tsx:116]: ``
```suggestion
                        {hostname != null
```

File: apps/journeys-admin/src/components/JourneyList/JourneyCard/JourneyCardMenu/DefaultMenu/DefaultMenu.tsx [hunk 15]
```diff
@@ -43,6 +45,14 @@ export function DefaultMenu({
   refetch
 }: DefaultMenuProps): ReactElement {
   const { t } = useTranslation('apps-journeys-admin')
+  const { activeTeam } = useTeam()
+  const { data: customDomainData } = useCustomDomainsQuery({
+    variables: { teamId: activeTeam?.id as string }
+  })
+
+  const hostname = new URL(`https://${customDomainData?.customDomains[0].name}`)
```
Comment (tataihono) [inline apps/journeys-admin/src/components/JourneyList/JourneyCard/JourneyCardMenu/DefaultMenu/DefaultMenu.tsx:53]:
```suggestion
  const hostname = customDomainData?.customDomains[0].name
```

File: apps/journeys-admin/src/components/JourneyList/JourneyCard/JourneyCardMenu/DefaultMenu/DefaultMenu.tsx [hunk 16]
```diff
@@ -68,7 +78,7 @@ export function DefaultMenu({
         />
       )}
       <NextLink
-        href={`/api/preview?slug=${slug}`}
+        href={`/api/preview?slug=${slug}&hostname=${hostname}`}
```
Comment (tataihono) [inline apps/journeys-admin/src/components/JourneyList/JourneyCard/JourneyCardMenu/DefaultMenu/DefaultMenu.tsx:81]: check if hostname is present if not then don't put in the arg

File: apps/journeys-admin/src/libs/useCustomDomainsQuery/useCustomDomainsQuery.tsx [hunk 17]
```diff
@@ -0,0 +1,42 @@
...
+): QueryResult<GetCustomDomains, GetCustomDomainsVariables> & {
+  hasCustomDomain: boolean
+} {
+  const query = useQuery<GetCustomDomains, GetCustomDomainsVariables>(
+    GET_CUSTOM_DOMAINS,
+    { ...options }
+  )
+
+  const hasCustomDomain =
+    query.data?.customDomains?.length !== 0 && query.data?.customDomains != null
```
Comment (tataihono) [inline apps/journeys-admin/src/libs/useCustomDomainsQuery/useCustomDomainsQuery.tsx:36]: drop

File: apps/journeys-admin/src/libs/useCustomDomainsQuery/useCustomDomainsQuery.tsx [hunk 18]
```diff
@@ -0,0 +1,42 @@
...
+
+  return {
+    ...query,
+    hasCustomDomain
+  }
```
Comment (tataihono) [inline apps/journeys-admin/src/libs/useCustomDomainsQuery/useCustomDomainsQuery.tsx:41]:
```suggestion
  return query
```

File: libs/locales/en/apps-journeys-admin.json [hunk 19]
```diff
@@ -601,6 +601,29 @@
...
+  "Domain name is a required field": "Domain name is a required field",
+  "Domain Settings": "Domain Settings",
+  "Default Journey": "Default Journey",
+  "Copied": "Copied",
+  "DNS Config": "DNS Config",
+  "Status": "Status",
+  "A": "A",
+  "CNAME": "CNAME",
+  "76.76.21.21": "76.76.21.21",
+  "cname.vercel-dns.com": "cname.vercel-dns.com",
```
Comment (tataihono) [inline libs/locales/en/apps-journeys-admin.json:620]: no translation required. we should remove this from the t function and add lint ignore.

**Review by tataihono — CHANGES_REQUESTED** [review]

**Review by tataihono — APPROVED** [review]

**Pull request merged**